### ISSUE TITLE
updating or adding new index to existing Dynamo table

### ISSUE THREAD

**Issue by Alua-Kinzhebayeva** [serverless/serverless#515]
Hey!
What is your opinion on adding a new index to an existing Dynamo table?
A bit of background. I have a table declared in s-module.json and already deployed. If I add a new index or even delete table from Dynamo console and try to deploy again with new index, CloudFormation won't let me deploy new table definition.
Is this use case a candidate for a plugin?

**Comment by c-knowles** [comment]
I think this may be more of a CloudFormation + DynamoDB question, although perhaps could be done with a plugin of some form.

From what I've experienced before, AWS doesn't always handle this as one may expect. It does say somewhere in the AWS docs about how it handles this but I failed to find it. Essentially on some resources if you have changed them enough from the stack definition then updates will fail since it can't perform the updates automatically.

Best docs I could find with a quick look today:
1. http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks.html
2. http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html - specifically the "Update requires" types on each attribute.

Are you adding a global or local secondary index? DynamoDB supports a "No interruption" update for global secondary indexes under some conditions - see the link from 2 above. Local secondary indexes require replacement unfortunately.

What I've done before is versioned the tables with a version at the end of the table name to be sure that when I spin up a changed schema it does not wipe out the previous table. Of course, often that does mean you do have to migrate the data and also your application will have to know about both versions for a migration period. I don't know of any other methods currently though.
